woodwork/table_accessor.py

```diff
@@ -541,6 +541,7 @@ def set_types(self, logical_types=None, semantic_tags=None, retain_index_tags=Tr
         for col_name, logical_type in logical_types.items():
             series = self._dataframe[col_name]
             updated_series = logical_type.transform(series)
+            logical_type.validate(updated_series)
```

Comment:
Is there sampling/head happening somewhere?

Reply:
No, I can add that